test/Api.Test/Controllers/AccountsControllerTests.cs

```diff
@@ -0,0 +1,351 @@
+                        .Returns(Task.FromResult(IdentityResult.Success));
+
+            await _sut.PostPassword(new PasswordRequestModel());
+
+            await _userService.Received(1).ChangePasswordAsync(user, default, default, default);
+        }
+
+        [Fact]
+        public async Task PostPassword_WhenNotAuthorized_ShouldThrowUnauthorizedAccessException()
+        {
+            ConfigureUserServiceToReturnNullPrincipal();
+
+            await Assert.ThrowsAsync<UnauthorizedAccessException>(
+                () => _sut.PostPassword(new PasswordRequestModel())
+            );
+        }
+
+        [Fact]
+        public async Task PostPassword_WhenPasswordChangeFails_ShouldBadRequestException()
+        {
+            var user = GenerateExampleUser();
+            ConfigureUserServiceToReturnValidPrincipalFor(user);
+            _userService.ChangePasswordAsync(user, default, default, default)
+                        .Returns(Task.FromResult(IdentityResult.Failed()));
+
+            await Assert.ThrowsAsync<BadRequestException>(
+                () => _sut.PostPassword(new PasswordRequestModel())
+            );
+        }
+
+        // Below are helper functions that currently belong to this
+        // test class, but ultimately may need to be split out into
+        // something greater in order to share common test steps with
+        // other test suites. They are included here for the time being
+        // until that day comes.
+        private User GenerateExampleUser()
+        {
+            return new User
+            {
+                Email = "user@example.com"
+            };
+        }
+
+        private void ConfigureUserServiceToReturnNullPrincipal()
+        {
+            _userService.GetUserByPrincipalAsync(Arg.Any<ClaimsPrincipal>())
+                        .Returns(Task.FromResult((User)null));
+        }
+
+        private void ConfigureUserServiceToReturnValidPrincipalFor(User user)
+        {
+            _userService.GetUserByPrincipalAsync(Arg.Any<ClaimsPrincipal>())
+                        .Returns(Task.FromResult(user));
+        }
+
+        private void ConfigureUserServiceToRejectPasswordFor(User user)
+        {
+            _userService.CheckPasswordAsync(user, Arg.Any<string>())
+                        .Returns(Task.FromResult(false));
+        }
+
+        private void ConfigureUserServiceToAcceptPasswordFor(User user)
+        {
+            _userService.CheckPasswordAsync(user, Arg.Any<string>())
+                        .Returns(Task.FromResult(true));
+        }
+
+        private void ConfigureUserServiceToReturnValidIdFor(User user)
+        {
+            _userService.GetUserByIdAsync(Arg.Any<Guid>())
+                        .Returns(Task.FromResult(user));
+        }
+
+        private void ConfigureUserServiceToReturnNullUserId()
+        {
+            _userService.GetUserByIdAsync(Arg.Any<Guid>())
+                        .Returns(Task.FromResult((User)null));
+        }
+    }
+}
```

Comment:
Please add a newline at the end of the code file.

Reply:
Newline added.